test/conformance/api/v1/util.go

```diff
@@ -128,8 +128,22 @@ func validateControlPlane(t pkgTest.T, clients *test.Clients, names test.Resourc
 		if r.Status.DeprecatedImageDigest == "" {
 			return false, fmt.Errorf("imageDigest not present for revision %s", names.Revision)
 		}
-		if validDigest, err := shared.ValidateImageDigest(names.Image, r.Status.DeprecatedImageDigest); !validDigest {
-			return false, fmt.Errorf("imageDigest %s is not valid for imageName %s: %w", r.Status.DeprecatedImageDigest, names.Image, err)
+		// for multi containers there will be more than one container and names.Image is a string,
+		// so in order to validate imageDigest for each images validateControlPlane accepting optional field called imagesForMultipleContainers
+		if names.Image == "" {
+			for _, value := range imagesForMultipleContainers {
+				for _, v := range r.Status.ContainerStatuses {
+					if image, ok := value[v.Name]; ok {
+						if validDigest, err := shared.ValidateImageDigest(image, v.ImageDigest); !validDigest {
+							return false, fmt.Errorf("imageDigest %s is not valid for imageName %s: %w", v.ImageDigest, image, err)
+						}
+					}
+				}
+			}
+		} else {
+			if validDigest, err := shared.ValidateImageDigest(names.Image, r.Status.DeprecatedImageDigest); !validDigest {
```

Comment:
if Im reading this right, this means we don't actually check r.Status.ContainerStatuses[0].ImageDigest is set for the single-container case? This seems a shame since the DeprecatedImageDigest field is deprecated.

Reply:
Yes still we are using `DeprecatedImageDigest` for single container.

I will send separate PR in order to remove that :+1:

Reply:
Modified as part of this commit https://github.com/knative/serving/pull/9387/commits/273aa43d32654f8954d9565b6a937366b022d0cf